docs/LifeOfAnInstruction.md

```diff
@@ -0,0 +1,202 @@
+# Machine code to bitcode: the life of an instruction
+
+This document describes how machine code instructions are lifted into LLVM bitcode. It should provide a detailed, inside-out overview of how McSema performs binary translation. This document omits some important steps, so consulting the [Navigating the code](NagivatingTheCode.md) document will be helpful.
+
+### Running example
+
+This document will use the instructions in the following basic block as a running example.
+
+![Sample basic block](images/instruction_life_block.png)
+
+## Step 1: CFG protocol buffer representation
+
+The first step to lifting involves getting the machine code instructions of some binary file into a format the McSema understands. We use a common and simple format so that McSema doesn't need to understand the many executable file formats ([ELF](https://en.wikipedia.org/wiki/Executable_and_Linkable_Format), [Mach-O](https://en.wikipedia.org/wiki/Mach-O), [PE](https://en.wikipedia.org/wiki/Portable_Executable), etc.).
+
+The file format used by McSema is a [CFG protocol buffer](/mcsema/CFG/CFG.proto). This file is produced using the [`mcsema-disass`](/mcsema/tools/mcsema_disass) tool. The CFG file for the above code block contains the following data.
+
+```protobuf
```

Comment:
I really like the breakdown; how do you generate this output from a protobuf file?